split.py

```diff
@@ -1,43 +1,110 @@
 import bpy
 import bl_operators
 import bmesh
+import math
 from math import ceil
+import numpy as np
 
 obj = bpy.context.active_object
 
-#colours
+# colours
 mat_zero = bpy.data.materials.new("mat_zero")
 mat_zero.diffuse_color = (0, 0, 0, 1)
 mat_one = bpy.data.materials.new("mat_one")
 mat_one.diffuse_color = (1, 0, 0, 1)
+mat_two = bpy.data.materials.new("mat_two")
+mat_two.diffuse_color = (0, 1, 0, 1)
+mat_three = bpy.data.materials.new("mat_three")
+mat_three.diffuse_color = (0, 0, 1, 1)
 
 obj.data.materials.append(mat_zero)
 obj.data.materials.append(mat_one)
+obj.data.materials.append(mat_two)
+obj.data.materials.append(mat_three)
 
 me = obj.data
 bm = bmesh.new()
 bm.from_mesh(me)
 
 
-#mark seam so that kerfed part stays in one piece
+def angle_between_normals(a, b):
+    dp = np.dot(a, b)
+    if (dp > 1):
+        dp = 1
+    return math.degrees(math.acos(dp))
+
+def remove_duplicates(duplicate):
+    final_list = []
+    for num in duplicate:
+        if num not in final_list:
+            final_list.append(num)
+    return final_list
+
+def remove_list(original, list):
+    final_list = []
+    for num in original:
+        if num not in list and num not in final_list:
+            final_list.append(num)
+    return final_list
+
+if hasattr(bm.verts, "ensure_lookup_table"):
+    bm.verts.ensure_lookup_table()
+    bm.edges.ensure_lookup_table()
+    bm.faces.ensure_lookup_table()
+
+# find seams and loops
+sharpest = 180
+seamedge = bm.edges[0]
+kerfedges = []
 for edge in bm.edges:
-    linked = edge.link_faces
-    if ((linked[0].smooth and not linked[1].smooth) or (not linked[0].smooth and linked[1].smooth)):
-        edge.seam = True
-        break
+
+    # find loops
+    faces = edge.link_faces
+    if (faces[0].normal.x == faces[1].normal.x):
```

Comment:
let's talk through this one as well—I'm trying to imagine, and it seems like the elif makes this incorrect, but I'd like to figure out why

Reply:
I'm happy to talk through this more! Essentially, if you hold any axis steady, the faces that rotate around it make up a loop

Reply:
NOTE: just colouring in for now, might do another kind of operation later